### PR TITLE
ENH: Synchronize ORC with upstream

### PR DESCRIPTION
**Pull request by bashtage**
Synchronize the orc read and write functions with upstream pandas

### CODE REVIEW

**Comment by bashtage** [comment]
No tests because it would require pyarrow, which is not installed.

File: pandas-stubs/io/orc.pyi [hunk 1]
```diff
@@ -1,7 +1,26 @@
...
+from pandas._typing import (
+    FilePath,
+    ReadBuffer,
+    WriteBuffer,
+)
 
 def read_orc(
-    path: FilePathOrBuffer, columns: list[str] | None = ..., **kwargs
+    path: FilePath | ReadBuffer[bytes],
+    columns: list[str] | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/io/orc.pyi:16]: Should this be `list[HashableT]` ?

File: pandas-stubs/io/orc.pyi [hunk 2]
```diff
@@ -1,7 +1,26 @@
...
+    **kwargs: Any,
 ) -> DataFrame: ...
+def to_orc(
```
Comment (Dr-Irv) [inline pandas-stubs/io/orc.pyi:19]: `pandas.io.to_orc()` is not documented, but v1.5 will have `DataFrame.to_orc()`, so can you put the stub there?

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

**Comment by Dr-Irv** [comment]
> No tests because it would require pyarrow, which is not installed.

It would just have to go in the section in `pyproject.toml` under `[tool.poetry.dev-dependencies]` so wouldn't be an installation requirement for the stubs, just for testing, and I think we should do that.

**Comment by Dr-Irv** [comment]
I see that you are skipping the tests if we have pandas < 1.5.  Did you test this with `main` from pandas?

File: pandas-stubs/io/orc.pyi (hunk 1 above)
Comment (bashtage) [reply pandas-stubs/io/orc.pyi]: Fixed.

**Comment by Dr-Irv** [comment]
I'm good with these changes, but can you answer this question: https://github.com/pandas-dev/pandas-stubs/pull/204#issuecomment-1220603356

Also need to resolve conflicts.  (realizing this is a PITA as I keep approving PR's)

**Comment by bashtage** [comment]
> I see that you are skipping the tests if we have pandas < 1.5. Did you test this with `main` from pandas?

Yes, it passes on main.

**Comment by bashtage** [comment]
Ready for merge.

**Review by Dr-Irv — APPROVED** [review]
thanks @bashtage

**Pull request merged**